codedoc.c

```diff
@@ -3117,8 +3117,10 @@ scan_file(filebuf_t   *file,		/* I  - File to scan */
 		    next_string = get_nth_text(type, 1, NULL);
 		  }
 		  else
+		  {
+        mxmlDelete(typedefnode);
```

Comment:
Deleting the typedef node isn't the right thing to do here - will do some more triage to see why the node isn't added to the output as expected.

Reply:
I would like to mention that in the case of the typedef node, it is being assigned to NULL thereafter (it cannot be added to output thereafter). Therefore it should be freed before assigned to NULL which does cause the leak with poc188.

Thank you for your time and reviewing the commit :)